pkg/network/macvtap.go

```diff
@@ -37,3 +37,12 @@ func renderMacvtapCni(conf *cnao.NetworkAddonsConfigSpec, manifestDir string, cl
 
 	return objs, nil
 }
+
+func fillMacvtapDefaults(conf *cnao.NetworkAddonsConfigSpec) []error {
+	// https://github.com/kubevirt/macvtap-cni/blob/be1528fb09e9ac3c490a5df31330851d7e1f8b0a/manifests/macvtap.yaml#L23
+	const defaultMacvtapDevicePluginConfigMapName = "macvtap-deviceplugin-config"
+	if conf.MacvtapCni != nil && conf.MacvtapCni.DevicePluginConfig == "" {
+		conf.MacvtapCni.DevicePluginConfig = defaultMacvtapDevicePluginConfigMapName
+	}
+	return nil
```

Comment:
No point in creating an empty slice when a nil does the same thing. 

The only difference would be when marshalling: the nil would marshall into `null` while the empty slice would marchall to `[]` . 

I think using nil is more idiomatic since that's the value it sets when you `var slice []<type>` . 

I can use your suggestion if you prefer.

Reply:
that's ok no need to change..